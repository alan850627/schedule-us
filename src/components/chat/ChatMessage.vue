<template>
  <div class="chat-message hanging-indent" :title="timeAgo">
    <b>{{ username }}</b>: {{ message }}
    <sup><small> {{timeAgo}}</small></sup>
  </div>
</template>

<script>
// import moment from 'moment'
// export default {
//   name: 'ChatMessage',

//   components: {
//   },

//   props: {
//     username: {
//       type: String,
//       default: 'Anonymous'
//     },
//     message: {
//       type: String,
//       default: 'Default Message.'
//     },
//     timestamp: {
//       type: String,
//       default: `${Date.now()}`
//     }
//   },

//   computed: {
//     timeAgo: function () {
//       return moment(Number(this.timestamp)).fromNow()
//     }
//   },

//   data () {
//     return {
//     }
//   },

//   methods: {
//   },

//   mounted () {
//   }
// }
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
.hanging-indent {
  padding-left: 25px ;
  text-indent: -25px ;
}
</style>
