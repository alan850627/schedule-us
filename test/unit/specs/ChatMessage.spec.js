import ChatMessage from '@/components/chat/ChatMessage'
import { shallow } from 'vue-test-utils'

describe('ChatMessage.vue', () => {
  it('has the expected html structure', () => {
    let cmp = shallow(ChatMessage, {
      propsData: {
        username: 'Alan',
        message: 'test message',
        timestamp: `${Date.now()}`
      }
    })
    expect(cmp.element).toMatchSnapshot()
  })
})
