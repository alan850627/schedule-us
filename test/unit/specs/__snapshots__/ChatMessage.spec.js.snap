// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage.vue has the expected html structure 1`] = `
<div
  class="chat-message hanging-indent"
  title="a few seconds ago"
>
  <b>
    Alan
  </b>
  : test message
  
  <sup>
    <small>
       a few seconds ago
    </small>
  </sup>
</div>
`;
